vm/src/obj/objstr.rs

```diff
@@ -33,7 +34,36 @@ impl<T: ToString> From<T> for PyString {
     }
 }
 
-pub type PyStringRef = PyRef<PyString>;
+pub trait IntoPyStringRef {
```

Comment:
Since this returns a `PyResult`, should it actually be called `TryIntoPyStringRef`?

Reply:
Sure. If it has to be ```try_into_pystringref``` this gets a bit unwieldy, but I guess there won't too many callers.

Reply:
Yeah, had the same concern for `TryIntoObject` (that's why I dropped the `Py` and `Ref`, since in practice you know it's referring to a `PyObject` from context and working with the value types instead of refs is pretty exceptional.

Reply:
Aha! The analogy to ```TryIntoObject``` is what I need to get the right shape of this. It's ```TryIntoRef<T>```. I this will be really useful eventually, and could replace ```TryFromObject```?

Reply:
How does this look now?

Reply:
> I this will be really useful eventually, and could replace TryFromObject?

`TryFromObject` is a bit different, e.g. it wouldn't make sense `TryIntoRef` for `isize`, but that is done for `TryFromObject` so we can accept it as a fn parameter. (Sorry if I caused confusion by typoing `TryFromObject` as `TryIntoObject` above).

But if my plan for `PyObjectRef` to eventually be a type alias for `PyRef<dyn PyObjectPayload>` works out, then this would prevent us from every _needing_ a `TryIntoObject` and `TryFromObject` could be replaced with a `TryFromRef`.

> How does this look now?

This looks great!

Reply:
I think creating ```impl TryIntoRef<PyInt> for isize``` does make sense.

They're sort of doing the same thing, in one case allowing us to write flexible parameters for functions that can be called from rust, in the other flexible paremeters for functions that can be called from python.

Reply:
Oh sorry that `isize` example was poorly thought out. I basically just meant that they are going in opposite directions:

`TryIntoRef`: Self -> Ref
`TryFromObject`: Ref -> Self

Reply:
I see now. They overlap in the one specific case where I've used TryFromObject (Self is PyRef), but there are other implementations.